File: frontend-app/src/types/share.ts
export interface Share {
  id: string;
  file_id: string;
  shared_by: string;
  shared_with_user_id: string | null;
  shared_with_email: string | null;
  share_token: string | null;
  permission: 'view' | 'edit' | 'admin';
  has_password: boolean;
  expires_at: string | null;
  created_at: string;
}

export interface SharedFile {
  file: {
    id: string;
    name: string;
    mime_type: string | null;
    size: number;
    is_folder: boolean;
  };
  permission: string;
  shared_by_name: string;
  shared_at: string;
}

export interface ShareCreateRequest {
  file_id: string;
  shared_with_email?: string;
  permission: string;
  create_link?: boolean;
  password?: string;
  expires_at?: string;
}

File: backend/app/schemas/share.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr


class ShareCreateRequest(BaseModel):
    file_id: uuid.UUID
    shared_with_email: EmailStr | None = None
    permission: str = "view"  # view, edit, admin
    create_link: bool = False
    password: str | None = None
    expires_at: datetime | None = None


class ShareResponse(BaseModel):
    id: uuid.UUID
    file_id: uuid.UUID
    shared_by: uuid.UUID
    shared_with_user_id: uuid.UUID | None = None
    shared_with_email: str | None = None
    share_token: str | None = None
    permission: str
    has_password: bool = False
    expires_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ShareUpdateRequest(BaseModel):
    permission: str | None = None  # view, edit, admin


class ShareLinkAccessRequest(BaseModel):
    password: str | None = None


class SharedFileResponse(BaseModel):
    file: dict
    permission: str
    shared_by_name: str
    shared_at: datetime
